environment.sh

```diff
@@ -36,52 +38,95 @@ done
 # Parse environment name
 env_name=$(basename "`pwd`")
 env_name+="_$env_type"
+branch_name=$(git rev-parse --abbrev-ref HEAD)
+# Determine which requirements.txt to install from
+if [ $env_type == 'simulation' ]; then
+  install_file="requirements.txt"
+elif [ $env_type == 'artifact' ]; then
+  install_file="artifact_requirements.txt"
+else
+  echo "Invalid environment type. Valid argument types are simulation and artifact."
+  return 
+fi
+requirements_modification_time="$(date -r $install_file '+%Y-%m-%d %H:%M:%S')"
+
+# Pull repo to get latest changes from remote if remote exists
+git ls-remote --exit-code --heads origin $branch_name >/dev/null 2>&1
+exit_code=$?
+if [[ $exit_code == '0' ]]; then
+  git fetch --all
+  echo "Git branch '$branch_name' exists in the remote repository"
+  git pull -u origin $branch_name
+fi
 
 # Check if environment exists already
-env_exists=$(conda info --envs | grep $env_name | head -n 1)
-if [[ $env_exists == '' ]]; then
-  env_exists="no"
+create_env=$(conda info --envs | grep $env_name | head -n 1)
+if [[ $create_env == '' ]]; then
+  # No environment exists with this name
+  create_env="yes"
 else
-  env_exists="yes"
+  create_env="no"
 fi
 
-if [ $env_exists == 'yes' ]; then
-  creation_time="$(head -n1 /home/$username/miniconda3/envs/$env_name/conda-meta/history)"
+if [ $create_env == 'no' ]; then
+  # Check if existing environment needs to be updated
   echo "Existing environment found for $env_name."
-  # Check if existing environment is older than a week and remake it if so
+  creation_time="$(head -n1 /home/$username/miniconda3/envs/$env_name/conda-meta/history)"
```

Comment:
Move `one_week_ago` definition down here so it's right by where you use it (and also don't need to calculate it unless you actually need it)